pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -679,6 +679,60 @@ func addExistsOperator(item model.TagQuery, tagMapType string, not bool) (string
 	return fmt.Sprintf(" AND %s (%s)", notStr, strings.Join(tagOperatorPair, " OR ")), args
 }
 
+func (r *ClickHouseReader) GetEntryPointOperations(ctx context.Context, queryParams *model.GetTopOperationsParams) (*[]model.TopOperationsItem, error) {
+	// Step 1: Get top operations for the given service
+	topOps, err := r.GetTopOperations(ctx, queryParams)
+	if err != nil {
+		return nil, err
+	}
+	if topOps == nil {
+		return nil, errors.New("no top operations found")
```

Comment:
Please use the new errors package in any new code you are adding.

you can import it as errorsv2